src/EnergyPlus/IOFiles.hh

```diff
@@ -100,7 +100,12 @@ public:
     void open(bool = false, bool = true);
     std::fstream::pos_type position() const noexcept;
 
-    void rewind() noexcept { if (is) { is->seekg(0); } }
+    void rewind() noexcept {
+        if (is) {
+            is->clear(); // clear eofbit and potentially failbit
+            is->seekg(0, std::ios::beg);
+        }
+    }
```

Comment:
And I suppose this is the fix for the weather manager hang.

Reply:
Yes, it is. Each commit is atomic to the bug fix it to attempting to solve.